File: GitToolBox/src/main/java/zielu/gittoolbox/extension/AutoFetchAllowed.java
package zielu.gittoolbox.extension;

import com.intellij.util.messages.Topic;

public interface AutoFetchAllowed {
  Topic<Notifier> TOPIC = Topic.create("Git ToolBox Auto Fetch Allowed", Notifier.class);

  void initialize();

  boolean isAllowed();

  interface Notifier {
    void stateChanged(AutoFetchAllowed allowed);
  }
}

File: GitToolBox/src/main/kotlin/zielu/gittoolbox/extension/autofetch/AutoFetchAllowed.kt
package zielu.gittoolbox.extension.autofetch

import com.intellij.openapi.project.Project
import com.intellij.util.messages.Topic

internal interface AutoFetchAllowed {
  fun isAllowed(project: Project): Boolean
}

internal interface AutoFetchAllowedNotifier {
  fun stateChanged()
}

internal val AUTO_FETCH_ALLOWED_TOPIC: Topic<AutoFetchAllowedNotifier> = Topic.create(
  "Git ToolBox Auto Fetch Allowed",
  AutoFetchAllowedNotifier::class.java
)
